grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1468,13 +1473,30 @@ class StockMovementService {
         Requisition requisition = stockMovement.requisition
         def shipment = requisition.shipment
 
+       validateRequisition(requisition)
+
         if (!shipment) {
             throw new IllegalStateException("There are no shipments associated with stock movement ${requisition.requestNumber}")
         }
 
         shipmentService.sendShipment(shipment, null, user, requisition.origin, stockMovement.dateShipped ?: new Date())
     }
 
+    void validateRequisition(Requisition requisition) {
+
+        requisition.requisitionItems.each { requisitionItem ->
+            if (!requisitionItem.picklistItems &&
```

Comment:
Looks pretty good. Add some comments to explain the logic. And let's make this a little more DRY by creating a `validateRequisitionItem(RequisitionItem)` just in case we need to reuse this at some point. 

Also I prefer the following syntax to make it more readable.
```
requisitionItem.status in [RequisitionItemStatus.CANCELED, ...]
```

Reply:
We need to test out a bunch of scenarios, so let's document as much as we can think of in the ticket. And I might want to add some unit tests to make sure we cover all of the scenarios we can think of.

Reply:
@jmiranda Sure, all required changes has been implemented. I'll also add comment in the ticket with all the scenarios I've tested while implementing this one.